client.go

```diff
@@ -476,6 +476,23 @@ func (a *Agent) JSON(v interface{}) *Agent {
 	return a
 }
 
+// ProblemJSON sends an RFC7807 JSON request.
+func (a *Agent) ProblemJSON(v interface{}) *Agent {
```

Comment:
don't really know if we should make separate functions for all json possibilities, then love a functionality that expects the json and a custom json type

Reply:
I agree. Since Go doesn't allow default parameter values or function overloading I've created a new method `CustomJSON` that accepts the JSON body and the content header as a string.

I'm wondering if this could result in any issues in the case of the user inputting an invalid ctype. Are there cases where the consumer would reject the message based on an invalid header and should we validate the header beforehand?